src/logd.rs

```diff
@@ -0,0 +1,117 @@
+use std::{io, os::unix::net::UnixDatagram, path::Path, time::SystemTime};
+
+use bytes::BufMut;
+use parking_lot::RwLockUpgradableReadGuard;
+
+use crate::{thread, Buffer, Event, Priority, LOGGER_ENTRY_MAX_LEN};
+
+/// Logd write socket path
+const LOGDW: &str = "/dev/socket/logdw";
+
+lazy_static::lazy_static! {
+    static ref SOCKET: LogdSocket = LogdSocket::connect(Path::new(LOGDW));
+}
+
+/// Logd write socket abstraction. Sends never fail and on each send a reconnect
+/// reconnect attempt is made.
+struct LogdSocket {
```

Comment:
This guy doesn't look to have logic specific to `logd`, maybe just `Socket` or `RetryConnectSocket` is more descriptive?